service/history/workflow/context.go

```diff
@@ -178,10 +178,15 @@ func NewContext(
 	throttledLogger log.ThrottledLogger,
 	metricsHandler metrics.Handler,
 ) *ContextImpl {
+	tags := []tag.Tag{
+		tag.WorkflowNamespaceID(workflowKey.NamespaceID),
+		tag.WorkflowID(workflowKey.WorkflowID),
+		tag.WorkflowRunID(workflowKey.RunID),
+	}
 	return &ContextImpl{
 		workflowKey:     workflowKey,
-		logger:          logger,
-		throttledLogger: throttledLogger,
+		logger:          log.With(logger, tags...),
```

Comment:
Probably create a lazy logger (`NewLazyLogger`), so we don't have to create a new logger, which won't be used in most cases, for every workflow.

Reply:
Neat, I didn't know we had that or that the overhead of the logger was so significant. 

Made the change 👍

Reply:
I am not 100% sure the logger overhead will be significant, but considering this is a very hot path and creating a real logger needs to copy tags, I think minimizing the overhead make sense here. Plus the change is also trivial :)